src/snyk/base/services/scanModeService.ts

```diff
@@ -1,47 +1,56 @@
+import { IConfiguration } from '../../common/configuration/configuration';
 import { EXECUTION_PAUSE_INTERVAL, EXECUTION_THROTTLING_INTERVAL } from '../../common/constants/general';
-import { SNYK_CONTEXT, SNYK_MODE_CODES } from '../../common/constants/views';
+import { SNYK_CONTEXT } from '../../common/constants/views';
 import { IContextService } from '../../common/services/contextService';
+import { CODE_SCAN_MODE } from '../../snykCode/constants/modes';
 
 export class ScanModeService {
-  private _mode = SNYK_MODE_CODES.AUTO;
+  private _mode = CODE_SCAN_MODE.AUTO;
   // Platform-independant type definition.
   private _unpauseTimeout: ReturnType<typeof setTimeout> | undefined;
   private _lastThrottledExecution: number | undefined;
 
-  constructor(private contextService: IContextService) {}
+  constructor(private contextService: IContextService, private config: IConfiguration) {}
 
-  isAutoScanAllowed(): boolean {
-    if ([SNYK_MODE_CODES.MANUAL, SNYK_MODE_CODES.PAUSED].includes(this._mode) || this.shouldBeThrottled()) {
+  isOssAutoScanAllowed(): boolean {
+    return this.config.shouldAutoScanOss;
+  }
+
+  isCodeAutoScanAllowed(): boolean {
```

Comment:
Don't sure. Maybe we could simplify this method? Like

`isCodeAutoScanAllowed() => ![CODE_SCAN_MODE.MANUAL, CODE_SCAN_MODE.PAUSED].includes(this._mode) || !this.shouldCodeBeThrottled()`